src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.targets

```diff
@@ -0,0 +1,240 @@
+  </PropertyGroup>
+
+  <!-- 
+    Gathers input assemblies for Tag Helper discovery and compilation. Add items to @(ReferencePath)
+  -->
+  <Target
+    Name="RazorResolveAssemblyReferences"
+    DependsOnTargets="ResolveReferences">
+    <ItemGroup>
+      <RazorReferencePath Include="@(ReferencePath)"/>
+      <RazorReferencePath Include="$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)$(TargetName)$(TargetExt)'))"/>
+    </ItemGroup>
+  </Target>
+
+  <Target
+    Name="_RazorResolveTagHelpers"
+    DependsOnTargets="CoreCompile;RazorResolveAssemblyReferences"
+    Inputs="$(MSBuildAllProjects);@(RazorReferencePath)"
+    Outputs="$(_RazorTagHelperInputCache)">
+
+    <!-- 
+      We're manipulating our output directly here because we want to separate the actual up-to-date check
+      of RazorCoreGenerate from the output of this target. Many times the set of tag helpers doesn't change
+      so we don't need to regenerate the code.
+      -->
+    <Touch
+      Files="$(_RazorTagHelperInputCache)"
+      AlwaysCreate="true">
+      <Output
+        TaskParameter="TouchedFiles"
+        ItemName="FileWrites" />
+    </Touch>
+
+    <RazorTagHelper
+      Debug="$(_RazorDebugTagHelperTask)"
+      DebugTool="$(_RazorDebugTagHelperTool)"
+      ToolAssembly="$(_RazorTagHelperToolAssembly)"
+      Assemblies="@(RazorReferencePath)"
+      TagHelperManifest="$(_RazorTagHelperOutputCache)">
+      <Output
+        TaskParameter="TagHelperManifest"
+        ItemName="FileWrites"/>
+    </RazorTagHelper>
+
+  </Target>
+
+  <Target Name="_RazorResolveSourceFiles">
+    <ItemGroup>
+      <RazorCompile Include="@(Content)" Condition="'%(Content.Extension)'=='.cshtml'" />
+      <_RazorGenerated Include="@(RazorCompile->'$(RazorGenerateOutputPath)%(RelativeDir)%(Filename).cs')">
+        <Source>%(Identity)</Source>
+      </_RazorGenerated>
+    </ItemGroup>
+  </Target>
+
+  <Target
+    Name="RazorCoreGenerate"
+    DependsOnTargets="_RazorResolveSourceFiles;_RazorResolveTagHelpers"
+    Inputs="$(MSBuildAllProjects);@(RazorCompile);$(_RazorTagHelperOutputCache)"
+    Outputs="@(_RazorGenerated)">
+
+    <RemoveDir
+      Directories="$(RazorGenerateOutputPath)"
+      Condition = "Exists('$(RazorGenerateOutputPath)')"/>
+
+    <MakeDir
+      Directories="%(_RazorGenerated.RelativeDir)"
+      Condition="!Exists('%(_RazorGenerated.RelativeDir)')" />
+
+    <RazorGenerate
+      Debug="$(_RazorDebugGenerateCodeTask)"
+      DebugTool="$(_RazorDebugGenerateCodeTool)"
+      ToolAssembly="$(_RazorGenerateToolAssembly)"
+      Sources="@(RazorCompile)"
+      ProjectRoot="$(MSBuildProjectDirectory)"
+      TagHelperManifest="$(_RazorTagHelperOutputCache)"
+      OutputPath="$(RazorGenerateOutputPath)" />
+
+    <ItemGroup>
+      <FileWrites Include="@(_RazorGenerated)" />
+    </ItemGroup>
+  </Target>
+
+  <!--
+    What follows here was copied and modified from the XamlPreCompile target in 
+    Microsoft.CSharp.CurrentVersion.targets
```

Comment:
I suspect that this will ultimately migrate to the CSharp targets. Xaml has its own call to CSC in a target that they own.

Reply:
https://github.com/aspnet/Razor/issues/1845